src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -685,39 +684,51 @@ WorkerShardStats(char *nodeName, uint32 nodePort, Oid relationId, char *shardNam
 		appendStringInfo(tableSizeQuery, SHARD_TABLE_SIZE_QUERY, quotedShardName);
 	}
 
-	queryOK = MultiClientExecute(connectionId, tableSizeQuery->data,
-								 &queryResult, &rowCount, &columnCount);
-	if (!queryOK)
+	querySent = SendRemoteCommand(connection, tableSizeQuery->data);
+	if (!querySent)
+	{
+		ReportConnectionError(connection, WARNING);
+		return false;
+	}
+
+	queryResult = GetRemoteCommandResult(connection, doRaiseInterrupts);
+	if (!IsResponseOK(queryResult))
 	{
-		MultiClientDisconnect(connectionId);
+		ReportResultError(connection, queryResult, WARNING);
+		PQclear(queryResult);
+		ForgetResults(connection);
 		return false;
 	}
 
-	tableSizeString = MultiClientGetValue(queryResult, 0, 0);
+	tableSizeString = PQgetvalue(queryResult, 0, 0);
 	if (tableSizeString == NULL)
 	{
-		MultiClientDisconnect(connectionId);
+		ereport(WARNING, (errmsg("received bad result from a shard placement, trying the"
```

Comment:
I've removed it but the idea was that without these messages, should this fail, we have no idea where it failed. I added slightly different verbiage to each of them so we would be able to take an error report and know exactly which condition was taken.

Without these messages we're silently discarding errors.